### PR TITLE
[EXTERNAL] add tosa erf and its lowering

### PR DESCRIPTION
**Pull request by manupak**
This commit adds tosa erf operator and its lowering
to math lib functions.

### CODE REVIEW

**Comment by manupak** [comment]
depends on https://github.com/ROCmSoftwarePlatform/rocMLIR/pull/1062

**Comment by manupak** [comment]
already started tosa discussion : https://discuss.mlplatform.org/t/adding-erf-operator-to-tosa/189/3.
lets see.

File: external/llvm-project/mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td [hunk 1]
```diff
@@ -390,6 +390,32 @@ def Tosa_TanhOp : Tosa_Op<"tanh", [
...
+// Operator: erf
+//===----------------------------------------------------------------------===//
+def Tosa_ErfOp : Tosa_Op<"erf", [
+    DeclareOpInterfaceMethods<InferShapedTypeOpInterface,
+                              ["inferReturnTypeComponents"]>,
+    Pure]> {
+  let summary = "Computes gauss error function of input";
+
+  let description = [{
+    Gauss error function.
```
Comment (krzysz00) [inline external/llvm-project/mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td]: Do we perhaps want to stick the definition of the gauss error function here to save people from having to run to a separate document?

File: external/llvm-project/mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td [hunk 2]
```diff
@@ -390,6 +390,32 @@ def Tosa_TanhOp : Tosa_Op<"tanh", [
...
+    For quantized integer data types, the TABLE operator should be used instead
```
Comment (krzysz00) [inline external/llvm-project/mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td:408]: Does `TABLE` already exist? And should it maybe be spelled `tosa.table`?

**Review by krzysz00 — COMMENTED** [review]
Add a test for the lowering, please, but looks good otherwise

File: external/llvm-project/mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td (hunk 1 above)
Comment (manupak) [reply external/llvm-project/mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td]: sure, I ll add that.

File: external/llvm-project/mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td (hunk 2 above)
Comment (manupak) [reply external/llvm-project/mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td:408]: yes it does. Im being consistent with other descriptions.

**Comment by manupak** [comment]
Its is kind of tested in the next PR (migraphx one) that suits our needs.
I thought to keep this as minimal as possible for now and the tests in upstream in the way they'd want it.

WDYT ? -- Im not against adding it here though, just wanted to know your thoughts.

**Comment by krzysz00** [comment]
I figured upstream will want the tests anyway, so might as well add them here. They probably have tests for all the other math functions, so you'll probably want to find those and add `erf` as a case.

**Comment by manupak** [comment]
OK I've added tests.

Also started upstreaming...
https://reviews.llvm.org/D150357
https://discuss.mlplatform.org/t/adding-erf-operator-to-tosa/189/9

**Review by krzysz00 — APPROVED** [review]

**Comment by manupak** [comment]
Thanks @krzysz00!

**Pull request merged**